### ISSUE TITLE
Possibility to customise checkboxes names in yii\grid\CheckboxColumn

### ISSUE THREAD

**Issue by SDKiller** [yiisoft/yii2#7810]
Now:

```
/**
  * @var string the name of the input checkbox input fields. This will be appended with `[]` to ensure it is an array.
  */
public $name = 'selection';
```

Use case: I need to submit multiple pairs of values (not such rare in practice).

Convenient way is to use input array keys, for example:

``` php
<input type="checkbox" 
           name="myName[<?= $model->id; ?>]" 
           value="<?= $model->someAttribute; ?>" />
```

Of course another way is to do it with `data` attributes, but you''ll need to specially handle them with js.

Now you cannot manage keys for autogenerated checkboxes in `CheckboxColumn` depending on current model, only values.
Braces `[]` are appended automatically.

Would be nice something like this:

``` php
'checkboxOptions' => function ($model, $key, $index, $column)  {

    return ['key' => $model->id, 'value' => $model->someAttribute];
},
```

**Comment by cebe** [comment]
related to #7806

**Comment by samdark** [comment]
Won't be implemented in 2.0.
